app/assets/javascripts/app/views/notifications_view.js

```diff
@@ -70,7 +70,8 @@ app.views.Notifications = Backbone.View.extend({
 
     [allNotes, typeNotes, headerBadge].forEach(function(element){
       element.text(function(i, text){
-        return parseInt(text) + change });
+        return Math.max(parseInt(text) + change, 0);
```

Comment:
Alright, I reverted this change. It is related to #5780 and I agree that one could interpret this as hiding the bug.